src/pages/Home/Home.tsx

```diff
@@ -25,70 +26,57 @@ export default function Home() {
   const [menuShow, setMenuShow] = useState(false);
   const [activeCategory, setActiveCategory] = useState("");
   const [categories, setCategories] = useState<CategoryPosition[]>([]);
-  const [scrollPosition, setScrollPosition] = useState(0);
-  const [loaded, setLoaded] = useState(false);
   const contentDivRef = useRef<HTMLDivElement>(null);
 
-  // onMount, add scrollListeners, Initialise categories, y-position, and default activeCategory.
   useEffect(() => {
     function scrollHandler() {
-      setScrollPosition(window.scrollY);
+      if (categories.length === 0) return; //guard against unloaded state;
+      const scrollPosition = window.scrollY;
+      let activeIndex = -1;
+      categories.forEach((category) => {
+        if (scrollPosition >= category.offsetY - HEADER_HEIGHT) {
+          activeIndex++;
+        }
+      });
+      setActiveCategory(categories[activeIndex].name);
```

Comment:
This is the scroll logic. 
It just loops through the offsetY and compares it with the scrollPosition.